gb test incorrectly passes flags to testing application

Simple project:
```
[nsf @ case]$ tree src
src
└── projectx
    └── main_test.go
```

Contents of `main_test.go`:

```go
package main

import (
	"flag"
	"testing"
)

var name = flag.String("name", "nsf", "what is your name")

func TestX(t *testing.T) {
	if *name != "jardin" {
		t.Fatalf("got: '%s', expected: 'jardin'", *name)
	}
}
```

Running `gb test`:
```
[nsf @ case]$ gb test -name=jardin
--- FAIL: TestX (0.00s)
        main_test.go:12: got: 'jardin=jardin', expected: 'jardin'
FAIL
FATAL command "test" failed: exit status 1
```

I'm using latest gb master as of now. 9c9026c

As you can see it passes flag as `<value>=<value>` instead of `<value>` for some reason.

--- a/cmd/gb/gb_test.go
+++ b/cmd/gb/gb_test.go
@@ -893,3 +893,33 @@ func TestTest(t *testing.T) {
 	gb.mustBeEmpty(tmpdir)
 	gb.mustNotExist(filepath.Join(gb.tempdir, "pkg")) // ensure no pkg directory is created
 }
+
+// https://github.com/constabulary/gb/issues/349
+func TestTestGbTestPassesUnknownFlags(t *testing.T) {
+	gb := T{T: t}
+	defer gb.cleanup()
+	gb.tempDir("src")
+	gb.tempDir("src/projectx")
+	gb.tempFile("src/projectx/main_test.go", `package main
+
+import (
+    "flag"
+    "testing"
+)
+
+var name = flag.String("name", "nsf", "what is your name")
+
+func TestX(t *testing.T) {
+    if *name != "jardin" {
+        t.Fatalf("got: '%s', expected: 'jardin'", *name)
+    }
+}
+`)
+	gb.cd(gb.tempdir)
+	tmpdir := gb.tempDir("tmp")
+	gb.setenv("TMP", tmpdir)
+	gb.run("test", "-name=jardin")
+	gb.grepStdout("^projectx$", "expected projectx") // output from gb test
+	gb.mustBeEmpty(tmpdir)
+	gb.mustNotExist(filepath.Join(gb.tempdir, "pkg")) // ensure no pkg directory is created
+}

--- a/cmd/gb/testflag.go
+++ b/cmd/gb/testflag.go
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+
+	"github.com/constabulary/gb/debug"
 )
 
 // testFlagSpec defines a flag we know about.
@@ -53,6 +55,7 @@ var testFlagDefn = map[string]*testFlagSpec{
 
 // TestFlags appends "-test." for flags that are passed to the test binary.
 func TestFlags(testArgs []string) []string {
+	debug.Debugf("TestFlags: args: %s", testArgs)
 	var targs []string
 	for _, arg := range testArgs {
 		var nArg, nVal, fArg string
@@ -70,14 +73,15 @@ func TestFlags(testArgs []string) []string {
 				}
 				if val.passToTest || val.passToAll {
 					fArg = "-test." + nArg
+					if nVal != "" {
+						fArg = fArg + "=" + nVal
+					}
 				}
-			}
-			if nVal != "" {
-				fArg = fArg + "=" + nVal
 			}
 		}
 		targs = append(targs, fArg)
 	}
+	debug.Debugf("testFlags: targs: %s", targs)
 	return targs
 }
 

--- a/cmd/gb/testflag_test.go
+++ b/cmd/gb/testflag_test.go
@@ -107,6 +107,13 @@ func TestTestFlagsPreParse(t *testing.T) {
 			args:  []string{"-P=1", "-short"},
 			pargs: []string{"-P=1"},
 			eargs: []string{"-short"},
+		}, {
+			args:  []string{"-name=jardin"},
+			eargs: []string{"-name=jardin"},
+		}, {
+			args:  []string{"-cover", "-name=jardin"},
+			pargs: []string{"-cover"},
+			eargs: []string{"-name=jardin"},
 		}}
 
 	for _, tt := range tests {

--- a/test/test.go
+++ b/test/test.go
@@ -70,6 +70,7 @@ func TestPackages(flags []string, pkgs ...*gb.Package) (*gb.Action, error) {
 // TestPackage returns an Action representing the steps required to build
 // and test this Package.
 func TestPackage(targets map[string]*gb.Action, pkg *gb.Package, flags []string) (*gb.Action, error) {
+	debug.Debugf("TestPackage: %s, flags: %s", pkg.ImportPath, flags)
 	var gofiles []string
 	gofiles = append(gofiles, pkg.GoFiles...)
 	gofiles = append(gofiles, pkg.TestGoFiles...)

